django/utils/decorators.py

```diff
@@ -111,13 +112,41 @@ def decorator_from_middleware(middleware_class):
     return make_middleware_decorator(middleware_class)()
 
 
+def sync_and_async_middleware(func):
```

Comment:
I'm confused by the definitions of these three functions here.  From [the docs](https://docs.djangoproject.com/en/3.1/topics/http/middleware/), it looks like these are already in 3.1.